run/filesystem/index.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const express = require('express');
+const rateLimit = require('express-rate-limit');
+const fs = require('fs');
+const serveIndex = require('serve-index');
+
+const app = express();
+const limit = rateLimit({
+  windowMs: 15 * 60 * 1000,
+  max: 100,
+  message: 'Rate limit exceeded',
+  headers: true,
+});
+const mntDir = process.env.MNT_DIR || '/mnt/nfs/filestore';
+const filePrefix = process.env.FILENAME || 'test';
+const port = parseInt(process.env.PORT) || 8080;
+
+app.use(limit);
+app.use('/filesystem', express.static(mntDir));
+app.use('/filesystem', serveIndex(mntDir));
+
+app.get('/', async (req, res) => {
+  await writeFile(mntDir);
+  let html =
+    '<html><body>A new file is generated each time this page is reloaded.<p>Files created on filesystem:<p>';
+  fs.readdirSync(mntDir).forEach(filename => {
+    html += `<a href="${req.protocol}://${req.get(
+      'host'
+    )}/filesystem/${filename}">${filename}</a><br>`;
+  });
+  html += '</body></html>';
+  res.send(html);
```

Comment:
Code was refactored to resolve the severe warnings from CodeQL. There remains 1 that is Medium.  I can refactor to resolve if it's okay that this sample's webpage looks different from its python and java counterparts.

Reply:
@averikitsch @pattishin any opinions on this?

Reply:
It looks like the input from the directory that's being read in needs to be validated and sanitized before using it is all to prevent injection attacks. Same treatment for the `request` coming in too since it looks like you're appending it in straight to the html. (go/xss)

Reply:
We should sanitize the input/output